src/flows/Onboarding/BasicInformationStep.tsx

```diff
@@ -2,18 +2,56 @@ import React from 'react';
 import { useOnboardingContext } from './context';
 import { OnboardingForm } from '@/src/flows/Onboarding/OnboardingForm';
 
+type BasicInformationFormPayload = {
+  name: string;
+  email: string;
+  work_email: string;
+  job_title: string;
+  tax_servicing_countries: string[];
+  tax_job_category: string;
+  department: {
+    id: string;
+    name?: string;
+  };
+  provisional_start_date: string;
+  has_seniority_date: 'yes' | 'no';
+  manager: {
+    id: string;
+  };
+  seniority_date: string;
+};
+
 type BasicInformationStepProps = {
   /*
    * The function is called when the form is submitted. It receives the form values as an argument.
    */
-  onSubmit?: (payload: unknown) => void | Promise<void>;
+  onSubmit?: (payload: BasicInformationFormPayload) => void | Promise<void>;
+  /*
+   * The function is called when an error occurs during form submission.
+   */
+  onError?: (error: unknown) => void;
 };
 
-export function BasicInformationStep({ onSubmit }: BasicInformationStepProps) {
+export function BasicInformationStep({
+  onSubmit,
+  onError,
+}: BasicInformationStepProps) {
   const { onboardingBag } = useOnboardingContext();
-  const handleSubmit = async (values: unknown) => {
-    await onSubmit?.(onboardingBag?.parseFormValues(values) as unknown);
-    onboardingBag?.next();
+  const handleSubmit = async (payload: Record<string, unknown>) => {
```

Comment:
Can't we set the payload type to `BasicInformationFormPayload`? We'd need to adjust `onSubmit` in `OnboardingForm` to accept an union of types